src/SamplesApp/UITests.Shared/Windows_UI_Xaml_Shapes/Basic_Shapes.xaml.cs

```diff
@@ -0,0 +1,651 @@
+								stretch.Alter(sut);
+								return sut;
+							},
+							$"[{shape.Name.ToUpperInvariant()}]\r\nSize: {size.Name}\r\nShape stretch: {stretch.Name}");
+						items.BorderBrush = new SolidColorBrush(Colors.HotPink);
+						items.BorderThickness = new Thickness(5);
+						stretchPanel.Children.Add(items);
+					}
+					sizePanel.Children.Add(stretchPanel);
+				}
+				shapesPanel.Children.Add(sizePanel);
+			}
+
+			_root.Visibility = Visibility.Visible;
+			RunningTest = "";
+			_testZone.Child = null;
+			_root.Content = shapesPanel;
+		}
+
+		private async void GenerateScreenshots(object sender, RoutedEventArgs e)
+		{
+#if WINDOWS_UWP
+			_root.Visibility = Visibility.Collapsed;
+
+			var folder = await new FolderPicker { FileTypeFilter = { "*" } }.PickSingleFolderAsync();
+			if (folder == null)
+			{
+				return;
+			}
+
+			var alteratorsMap = _stretches.SelectMany(stretch => _sizes.Select(size => new[] { stretch, size })).ToArray();
+
+			foreach (var shape in _shapes)
+			foreach (var alterators in alteratorsMap)
+			{
+				var fileName = shape.Name + "_" + string.Join("_", alterators.Select(a => a.Id)) + ".png";
+				var grid = BuildHoriVertTestGridForScreenshot(shape, alterators);
+				_testZone.Child = grid;
+				await Task.Yield();
+
+				var renderer = new RenderTargetBitmap();
+				await renderer.RenderAsync(grid);
+
+				var file = await folder.CreateFileAsync(fileName, CreationCollisionOption.ReplaceExisting);
+				using (var output = await file.OpenAsync(FileAccessMode.ReadWrite))
+				{
+					var encoder = await BitmapEncoder.CreateAsync(BitmapEncoder.PngEncoderId, output);
+					encoder.SetSoftwareBitmap(SoftwareBitmap.CreateCopyFromBuffer(await renderer.GetPixelsAsync(), BitmapPixelFormat.Bgra8, renderer.PixelWidth, renderer.PixelHeight));
+					await encoder.FlushAsync();
+					await output.FlushAsync();
+				}
+			}
+
+			_root.Visibility = Visibility.Visible;
+			_testZone.Child = null;
+#endif
+		}
+
+#if __IOS__ // This is the base work for a fix in the UITests in order to highly increase Screenshot speed. It will be removed by https://github.com/unoplatform/Uno.UITest/issues/39
+		private static byte[] RenderAsPng(FrameworkElement elt)
+		{
+			UIImage img;
+			try
+			{
+				UIGraphics.BeginImageContextWithOptions(new Size(elt.ActualWidth, elt.ActualHeight), true, UIScreen.MainScreen.Scale);
+				var ctx = UIGraphics.GetCurrentContext();
+				ctx.SetFillColor(Colors.White);
+				elt.Layer.RenderInContext(ctx);
+				img = UIGraphics.GetImageFromCurrentImageContext();
+			}
+			finally
+			{
+				UIGraphics.EndImageContext();
+			}
+
+			using (img)
+			{
+				return img.AsPNG().ToArray();
+			}
+		}
+#else
+		private static byte[] RenderAsPng(FrameworkElement elt)
+			=> throw new NotImplementedException("Not supported yet on this platform");
+#endif
+
+		public string RunTests(string testNames)
+		{
+			TestResult = "";
+
+			var tests = testNames.Split(new [] {';'}, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'testNames' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5633443)